src/controllers/scripting/legacy/controllerscriptinterfacelegacy.cpp

```diff
@@ -1052,3 +1058,288 @@ void ControllerScriptInterfaceLegacy::softStart(int deck, bool activate, double
+    case WellKnownCharsets::IBM1026:
+        return convertCharsetInternal(QStringLiteral("IBM1026"), value);
+    case WellKnownCharsets::IBM1047:
+        return convertCharsetInternal(QStringLiteral("IBM1047"), value);
+    case WellKnownCharsets::IBM01140:
+        return convertCharsetInternal(QStringLiteral("IBM01140"), value);
+    case WellKnownCharsets::IBM01141:
+        return convertCharsetInternal(QStringLiteral("IBM01141"), value);
+    case WellKnownCharsets::IBM01142:
+        return convertCharsetInternal(QStringLiteral("IBM01142"), value);
+    case WellKnownCharsets::IBM01143:
+        return convertCharsetInternal(QStringLiteral("IBM01143"), value);
+    case WellKnownCharsets::IBM01144:
+        return convertCharsetInternal(QStringLiteral("IBM01144"), value);
+    case WellKnownCharsets::IBM01145:
+        return convertCharsetInternal(QStringLiteral("IBM01145"), value);
+    case WellKnownCharsets::IBM01146:
+        return convertCharsetInternal(QStringLiteral("IBM01146"), value);
+    case WellKnownCharsets::IBM01147:
+        return convertCharsetInternal(QStringLiteral("IBM01147"), value);
+    case WellKnownCharsets::IBM01148:
+        return convertCharsetInternal(QStringLiteral("IBM01148"), value);
+    case WellKnownCharsets::IBM01149:
+        return convertCharsetInternal(QStringLiteral("IBM01149"), value);
+    default:
+        m_pScriptEngineLegacy->logOrThrowError(QStringLiteral("Unknown charset specified"));
+        return QByteArray();
+    }
+}
+
+#if QT_VERSION < QT_VERSION_CHECK(6, 4, 0)
+// Add Byte Order Mark (BOM) to the beginning of the data if it is not already present
+static QByteArray addBom(const QByteArray& data) {
+    static const QByteArray bomBE = QByteArray::fromHex("FEFF");
+    static const QByteArray bomLE = QByteArray::fromHex("FFFE");
+
+    // Check system endianness
+    if (QSysInfo::ByteOrder == QSysInfo::BigEndian) {
+        if (!data.startsWith(bomBE)) {
+            return bomBE + data;
+        }
+    } else {
+        if (!data.startsWith(bomLE)) {
+            return bomLE + data;
+        }
+    }
+    return data;
+}
+#endif
+
+QByteArray ControllerScriptInterfaceLegacy::convertCharsetInternal(
+        const QString& targetCharset, const QString& value) {
+#if QT_VERSION < QT_VERSION_CHECK(6, 4, 0)
+    QByteArray encoderNameArray = targetCharset.toUtf8();
+    auto* pCodec = QTextCodec::codecForName(encoderNameArray);
+    if (!pCodec) {
+        m_pScriptEngineLegacy->logOrThrowError(
+                QStringLiteral("Unable to get QTextCodec name for charset: %1").arg(targetCharset));
+        return QByteArray();
+    }
+    std::unique_ptr<QTextEncoder> encoder(
+            pCodec->makeEncoder(QTextCodec::Flag::ConvertInvalidToNull));
+    if (encoderNameArray == "ISO-10646-UCS-2" || encoderNameArray == "UCS2") {
+        // For these encodings QTextCodec does not set, the BOM which QStringEncoder does
+        return addBom(encoder->fromUnicode(value));
+    }
```

Comment:
it would be nice to expose `QStringConverter::Flag::WriteBom` as in I found in practice that the BOM is rarely used in controller protocols.

Reply:
Here it's very relevant, as we send strings encoded in multi-byte charsets to a controller with endianess known at development time of the mapping script, but from a host computer with endianess unknown at time of mapping development.

Reply:
So the endianes of a string encoded in UCS2 always matches that of the host?

Reply:
I think so, but don't know for sure. The same applies for most other charsets, but for all the other multi-byte charsets there was no difference between the output of `QTextCodec` and `QStringEncoder`.
On the other hand, the number of users running Mixxx on a BigEndian CPU architecture is presumably negligible.

Reply:
likely yeah, but it would be nice to confirm whats the case. The mapping creator needs to be certain what endianness their returned string is...